flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/extraction/FunctionMappingExtractor.java

```diff
@@ -174,6 +178,39 @@ boolean hasAccumulator() {
 		return collectedMappings;
 	}
 
+	/**
+	 * Special case for Scala which generates two methods when using var-args (a {@code Seq < String >}
+	 * and {@code String...}). This method searches for the Java-like variant.
+	 */
+	private static Method adjustMethod(Method method) {
```

Comment:
This could be called sth like `correctVarargMethod()` that describes what it does in the method name.